lib/compat/wordpress-6.5/fonts/class-wp-font-utils.php

```diff
@@ -28,30 +28,26 @@ class WP_Font_Utils {
 		 * @link https://www.w3.org/TR/css-fonts-4/#font-family-prop
 		 *
 		 * @since 6.5.0
-		 * @access private
-		 *
-		 * @see sanitize_font_family()
 		 *
 		 * @param string $item A font family name.
-		 * @return string The font family name with surrounding quotes if necessary.
+		 * @return string The font family name with surrounding quotes, if necessary.
 		 */
 		private static function maybe_add_quotes( $item ) {
-			// Match any non alphabetic characters (a-zA-Z), dashes -, or parenthesis ().
-			$regex = '/[^a-zA-Z\-()]+/';
+			// Matches strings that are not exclusively alphabetic characters or hyphens, and do not exactly follow the pattern generic(alphabetic characters or hyphens).
+			$regex = '/^(?!generic\([a-zA-Z\-]+\)$)(?!^[a-zA-Z\-]+$).+/';
 			$item  = trim( $item );
 			if ( preg_match( $regex, $item ) ) {
```

Comment:
there's a case in this regex that can be detected correctly:
`generic(font` will match it because it's not considering the trailing `)`.

Reply:
Apart from that, making the regex positive instead of negative makes it less specific and could match other unwanted strings, for example, an empty string. For those reasons I think we should keep the current regex.

Reply:
Good points, thanks for explaining. I see why the negative matching makes more sense in this case.